test/integration/BCowPool.t.sol

```diff
@@ -0,0 +1,181 @@
+      validTo: latestValidTimestamp,
+      appData: APP_DATA,
+      feeAmount: 0,
+      kind: GPv2Order.KIND_BUY,
+      partiallyFillable: false,
+      buyTokenBalance: GPv2Order.BALANCE_ERC20,
+      sellTokenBalance: GPv2Order.BALANCE_ERC20
+    });
+
+    // swapper signs the order
+    (uint8 v, bytes32 r, bytes32 s) =
+      vm.sign(swapper.privateKey, GPv2Order.hash(swapperOrder, settlement.domainSeparator()));
+    bytes memory swapperSig = abi.encodePacked(r, s, v);
+
+    // order for bPool is generated
+    GPv2Order.Data memory poolOrder = GPv2Order.Data({
+      sellToken: dai,
+      buyToken: weth,
+      receiver: GPv2Order.RECEIVER_SAME_AS_OWNER,
+      sellAmount: buyAmount, // 100 DAI
+      buyAmount: sellAmount, // WETH
+      validTo: latestValidTimestamp,
+      appData: APP_DATA,
+      feeAmount: 0,
+      kind: GPv2Order.KIND_SELL,
+      partiallyFillable: true,
+      sellTokenBalance: GPv2Order.BALANCE_ERC20,
+      buyTokenBalance: GPv2Order.BALANCE_ERC20
+    });
+    bytes memory poolSig = abi.encode(poolOrder);
+
+    // solver prepares for call settle()
+    IERC20[] memory tokens = new IERC20[](2);
+    tokens[0] = IERC20(weth);
+    tokens[1] = IERC20(dai);
+
+    uint256[] memory clearingPrices = new uint256[](2);
+    // TODO: we can use more accurate clearing prices here
+    clearingPrices[0] = buyAmount;
+    clearingPrices[1] = sellAmount;
+
+    GPv2Trade.Data[] memory trades = new GPv2Trade.Data[](2);
+
+    // pool's trade
+    trades[0] = GPv2Trade.Data({
+      sellTokenIndex: 1,
+      buyTokenIndex: 0,
+      receiver: poolOrder.receiver,
+      sellAmount: poolOrder.sellAmount,
+      buyAmount: poolOrder.buyAmount,
+      validTo: poolOrder.validTo,
+      appData: poolOrder.appData,
+      feeAmount: poolOrder.feeAmount,
+      flags: GPv2TradeEncoder.encodeFlags(poolOrder, GPv2Signing.Scheme.Eip1271),
+      executedAmount: poolOrder.sellAmount,
+      signature: abi.encodePacked(address(pool), poolSig)
+    });
+
+    // swapper's trade
+    trades[1] = GPv2Trade.Data({
+      sellTokenIndex: 0,
+      buyTokenIndex: 1,
+      receiver: swapperOrder.receiver,
+      sellAmount: swapperOrder.sellAmount,
+      buyAmount: swapperOrder.buyAmount,
+      validTo: swapperOrder.validTo,
+      appData: swapperOrder.appData,
+      feeAmount: swapperOrder.feeAmount,
+      flags: GPv2TradeEncoder.encodeFlags(swapperOrder, GPv2Signing.Scheme.Eip712),
+      executedAmount: swapperOrder.sellAmount,
+      signature: swapperSig
+    });
+
+    // in the first interactions, save the commitment
+    GPv2Interaction.Data[][3] memory interactions =
+      [new GPv2Interaction.Data[](1), new GPv2Interaction.Data[](0), new GPv2Interaction.Data[](0)];
+
+    interactions[0][0] = GPv2Interaction.Data({
+      target: address(pool),
+      value: 0,
+      callData: abi.encodeWithSelector(
+        IBCoWPool.commit.selector, poolOrder.hash(pool.SOLUTION_SETTLER_DOMAIN_SEPARATOR())
+      )
+    });
```

Comment:
hmmmmm interesting, i just worry about: what avoids the settler from adding
```solidity
  interactions[0][0] = GPv2Interaction.Data({
    target: address(DAI),
    value: 0,
    callData: abi.encodeWithSelector(
      ERC20.transferFrom.selector, address(pool), address(bunny), 100000
    )
```

Reply:
ahá! the `solutionSettler` contract doesn't have the allowances :) the `vaultRelayer` has